zerver/lib/notifications.py

```diff
@@ -71,6 +71,17 @@ def topic_narrow_url(realm, stream, topic):
 def relative_to_full_url(base_url, content):
     # type: (Text, Text) -> Text
     # Convert relative URLs to absolute URLs.
+    # We handle narrow URLs separately because of two reasons:
+    # 1: 'lxml' seems to be having an issue in dealing with URLs that begin
+    # `#` due to which it doesn't add a `/` before joining the base_url to
+    # the relative URL.
+    # 2: We also need to update the title attribute in the narrow links which
+    # is not possible with `make_links_absolute()`.
+    content = re.sub(
+        r"(?<=\=['\"])(/{0,1}#narrow/)(?=[^<]+>)",
+        base_url + r"/#narrow/",
+        content)
+
```

Comment:
This is probably OK for a quick fix, but I think we probably want to do something more like `get_text_content`, by which I mean having `lxml` parse the content and only modify links are inside a link `href` tag, both for this and file uploads item a bit below.  That way, we can not be doing anything with hacky regular expressions.  

Might be worth trying that better approach to see how hard it is first?  Would basically mean implementing our own slightly fancier version of `make_links_absolute`, I think...